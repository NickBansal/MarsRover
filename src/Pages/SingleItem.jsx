import React, { Component } from 'react'
import * as api from '../api';
import '../Stylesheets/SingleItem.css'
import Loading from '../Components/Loading'
import NotAvailable from '../Stylesheets/Images/not-available.png';
import moment from 'moment'

class SingleItem extends Component {

    state = {
        singleItem: {},
        error: false,
        loading: true,
        assets: []
    }

    render() {
        const { loading, singleItem, assets } = this.state
        return (
            <div>
                {loading && <Loading />}
                {!loading &&
                    <div style={{ paddingTop: '40px' }}>
                        {singleItem.data[0].media_type === 'image' && <div className="hvrbox">
                            <img
                                onError={(e) => this.addDefaultSrc(e)}
                                src={singleItem.links[0].href}
                                alt={singleItem.data[0].title}
                                className="hvrbox-layer_bottom" />
                            <div className="hvrbox-layer_top hvrbox-layer_scale">
                                <div className="hvrbox-text">
                                    <h1>{singleItem.data[0].title}</h1>
                                    <p className="Description">{singleItem.data[0].description}</p>
                                    <p className="Created">Created: {moment(singleItem.data[0].date_created).from()}</p>
                                </div>
                            </div>
                            <p>Please hover over the image for more detail</p>
                        </div>}
                        {singleItem.data[0].media_type === 'video' &&
                            <div className="VideoPlayback">
                                <video width="550" height="550" controls>
                                    <source src={assets[0].href} type="video/mp4" />
                                    <source src={assets[1].href} type="video/mp4" />
                                </video>
                                <h1>{singleItem.data[0].title}</h1>
                                <p className="Description">{singleItem.data[0].description}</p>
                                <p className="Created">Created: {moment(singleItem.data[0].date_created).from()}</p>
                            </div>
                        }
                    </div>
                }
            </div>
        )
    }

    addDefaultSrc = (event) => {
        event.target.src = `${NotAvailable}`
    }

    componentDidMount() {
        const getItems = api.getItems(this.props.id)
        const getAssets = api.assetData(this.props.id)
        Promise.all([getItems, getAssets])
            .then(data => {
                this.setState({
                    singleItem: data[0].collection.items[0],
                    loading: false,
                    assets: data[1].collection.items
                })
            })
            .catch(error => {
                this.setState({
                    error: true
                })
            })
    }
}

export default SingleItem