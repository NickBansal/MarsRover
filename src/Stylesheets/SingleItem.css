.SingleItem {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center
}

.hvrbox,
.hvrbox * {
	-webkit-box-sizing: border-box;
	        box-sizing: border-box;
}
.hvrbox {
	position: relative;
	display: inline-block;
	overflow: hidden;
	max-width: 100%;
    height: auto;
    border-radius: 20px;
}
.hvrbox img {
    width: 550px;
    height: 550px;
    border: 5px solid white;
    border-radius: 20px;
    -webkit-box-shadow: 0px 9px 45px 0px rgba(0,0,0,0.75);
    -moz-box-shadow: 0px 9px 45px 0px rgba(0,0,0,0.75);
    box-shadow: 0px 9px 45px 0px rgba(0,0,0,0.75); 
}
.hvrbox .hvrbox-layer_bottom {
    display: block;
    border-radius: 20px;
    opacity: 0;
    animation: 1 fadein 1.5s forwards
}

@keyframes fadein {
    100% {
        opacity: 1;
    }
}

.hvrbox .hvrbox-layer_top {
	opacity: 0;
    position: absolute;
    border-radius: 20px;
	left: 0;
	right: 0;
	top: 0;
	height: 550px;
    width: 550px;
	background: rgba(0, 0, 0, 0.8);
	border: 4px solid white; 
	color: #fff;
	padding: 15px;
	-webkit-transition: all 0.4s ease-in-out 0s;
	-o-transition: all 0.4s ease-in-out 0s;
	transition: all 0.4s ease-in-out 0s;
	overflow: scroll
}
.hvrbox:hover .hvrbox-layer_top,
.hvrbox.active .hvrbox-layer_top {
	opacity: 1;
	cursor: pointer;
	padding: 20px
}
.hvrbox .hvrbox-text {
	text-align: center;
	font-size: 18px;
	display: inline-block;
}

.hvrbox .hvrbox-text_mobile {
	font-size: 15px;
	border-top: 1px solid rgb(179, 179, 179); /* for old browsers */
	border-top: 1px solid rgba(179, 179, 179, 0.7);
	margin-top: 5px;
	padding-top: 2px;
	display: none;
}
.hvrbox.active .hvrbox-text_mobile {
	display: block;
}

.hvrbox .hvrbox-layer_scale {
	border-radius: 50%;
	-webkit-transform: scale(0);
	-ms-transform: scale(0);
	transform: scale(0);
}
.hvrbox:hover .hvrbox-layer_scale,
.hvrbox.active .hvrbox-layer_scale {
	border-radius: 20px;
	-webkit-transform: scale(1);
	-ms-transform: scale(1);
	transform: scale(1);
}

.hvrbox-text h1 {
    font-size: 35px;
    margin: 0
}

.hvrbox-text p.Description {
	font-size: 20px
}

.hvrbox-text p.Created {
	font-size: 15px
}

video {
	height: 310px;
	width: 545px;
	border-radius: 20px;
	border: 3px solid white;
	background: #0000005e
}

.fa-chevron-circle-left {
	color: white;
	transition: 0.3s
}

.fa-chevron-circle-left:hover {
	cursor: pointer;
	transform: translateY(-2px);
	color: lightgray;
	transform: scale(1.2);
	transition: 0.2s
}

.fa-5x {
	font-size: 5em;
}

button {
	background: none;
	outline: none;
	border: none
}

.VideoPlayback {
	opacity: 0;
	animation: 1 fadein 1.5s forwards
}

.VideoPlayback p {
	width: 60%;
	margin: 15px auto
}

.VideoPlayback h1 {
	width: 60%;
	margin: 20px auto
}

.ErrorLoad {
	font-size: 25px
}

@media (max-width: 850px) {
	.VideoPlayback p {
		width: 80%;
	}
	.VideoPlayback h1 {
		width: 80%;
	}
}

@media (max-width: 600px) {
	.hvrbox .hvrbox-layer_top {
		height: 300px;
		width: 300px;
	}
	.hvrbox img {
		height: 300px;
		width: 300px;
	}
	.hvrbox-text h1 {
		font-size: 20px;
	}
	.hvrbox-text p.Description {
		font-size: 15px
	}
	.hvrbox-text p.Created {
		font-size: 13px
	}
	video {
		height: 175px;
		width: 300px
	}
	.VideoPlayback p {
		width: 100%;
		font-size: 15px
	}
	.VideoPlayback h1 {
		font-size: 22px
	}
	.ErrorLoad {
		font-size: 20px
	}
	.fa-5x {
		font-size: 3em;
	}
}