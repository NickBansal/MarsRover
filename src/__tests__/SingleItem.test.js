import './setupTest';
import React from 'react';
import SingleItem from '../Pages/SingleItem';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import axios from 'axios'

describe('<LandingPage />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<SingleItem />)
    })
    it('Component matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('componentDidMount should fetch items and update the state', async () => {
        const renderedComponent = await wrapper
        await renderedComponent.update()
        expect(renderedComponent.state('groceries').length).toEqual(2)
      })
})