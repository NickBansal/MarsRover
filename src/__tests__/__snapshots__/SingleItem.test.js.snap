// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> Component matches the snapshot 1`] = `
<div>
  <Loading />
</div>
`;
